packages/babel-plugin-transform-async-to-promises/src/index.js

```diff
@@ -15,6 +15,32 @@ function parse(code) {
   }).program;
 }
 
+/* Polyfill for Object.assign(...) from: 
```

Comment:
Please do not use MDN code in production anywhere :-/ this code doesn’t work with symbols, and has a number of other subtle bugs.

I’d suggest `object.assign` on npm.

Reply:
Well we don't need this at all, can just use `Object.assign` because min version is Node 4, or object spread

Reply:
If you look closely, it's not polyfilling the prototype, it's just defined & called locally. In fact, it could easily be replaced by a simple for-loop. And as hzoo says, if you're only targeting Node >4, Object.assign() will work fine.

@hzoo - can I assume you're OK to maintain this for now?

Reply:
Babel 7 does target Node 4 so you can use Object.assign by itself in this PR

Reply:
@MatAtBread right, i'm saying that basically all the code on MDN is broken and should never be used for any reason beyond "loosely explaining what a thing does". Since babel requires node 4+, this can just be deleted, but the larger point about MDN code stands.